source/tests/system/nidcpower_driver_api_tests.cpp

```diff
@@ -140,6 +238,117 @@ TEST_F(NiDCPowerDriverApiTest, NiDCPowerReset_SendRequest_ResetCompletesSuccessf
   expect_api_success(response.status());
 }
 
+TEST_F(NiDCPowerDriverApiTest, NiDCPowerSetViInt32Attribute_SendRequest_GetViInt32AttributeMatches)
+{
+  const char* channel_name = "";
+  const dcpower::NiDCPowerAttributes attribute_to_set = dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_MEASURE_WHEN;
+  const ViInt32 expected_value = dcpower::MeasureWhen::MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE;
+  ::grpc::ClientContext context;
+  dcpower::SetAttributeViInt32Request request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_name);
+  request.set_attribute_id(attribute_to_set);
+  request.set_attribute_value(expected_value);
+  dcpower::SetAttributeViInt32Response response;
+
+  ::grpc::Status status = GetStub()->SetAttributeViInt32(&context, request, &response);
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+  
+  ViInt32 get_attribute_value = get_int32_attribute(channel_name, attribute_to_set);
+  EXPECT_EQ(expected_value, get_attribute_value);
+}
+
+TEST_F(NiDCPowerDriverApiTest, NiDCPowerSetViReal64Attribute_SendRequest_GetViReal64AttributeMatches)
+{
+  const char* channel_name = "0";
+  // Attribute 'NIDCPOWER_ATTRIBUTE_MEASURE_WHEN' must be set to 'MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE' 
+  // before setting attribute 'NIDCPOWER_ATTRIBUTE_SOURCE_DELAY'.
+  set_int32_attribute(channel_name, dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_MEASURE_WHEN, dcpower::MeasureWhen::MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE);
```

Comment:
This line's getting pretty long. Can you break the arguments onto separate lines for readability? Also, do the same for the other tests setting this attribute.

Reply:
Done. Now it will look like this. 
set_int32_attribute(
    channel_name, 
    dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_MEASURE_WHEN, 
    dcpower::MeasureWhen::MEASURE_WHEN_NIDCPOWER_VAL_AUTOMATICALLY_AFTER_SOURCE_COMPLETE
);